### ISSUE TITLE
Replacing the okhttp based sender using JDK autoconfigure

### ISSUE THREAD

**Issue by fax4ever** [open-telemetry/opentelemetry-java#5958]
**Is your feature request related to a problem? Please describe.**

We would like to replace the okhttp based sender. It seems to be possible replacing the usages of `opentelemetry-exporter-sender-okhttp` with `opentelemetry-exporter-sender-jdk` and `opentelemetry-exporter-sender-grpc-managed-channel`.

All work if create the SDK manually and setting the channel:

``` java
ManagedChannel managedChannel = ManagedChannelBuilder.forAddress("localhost", 4317).usePlaintext().build();
OtlpGrpcSpanExporter otlpGrpcExporter = OtlpGrpcSpanExporter.builder()
            .setChannel(managedChannel)
            .build();
``` 

What if the OpenTelemetry SDK is created by the autoconfigure project?

**Describe the solution you'd like**

Find a way to replace the okhttp based sender with the SDK autoconfigure.

**Describe alternatives you've considered**

I tried to add a `SpanExporterCustomizer` without success.

**Additional context**

Hi tried to do it here: https://github.com/fax4ever/telemetry-play/blob/main/sdk-autoconfigure-exporter/src/main/java/fax/play/OTelConfig.java.

**Comment by jack-berg** [comment]
> It seems to be possible replacing the usages of opentelemetry-exporter-sender-okhttp with opentelemetry-exporter-sender-jdk and opentelemetry-exporter-sender-grpc-managed-channel.

That's correct. Based on your source code, you're using the gRPC version. The only alternative to the okhttp implementation for gRPC is `opentelemetry-exporter-sender-grpc-managed-channel`. To use, include the artifact (and exclude the `opentelemetry-exporter-sender-okhttp`, then call `setChannel`.

If you want to do this with autoconfigure, you'll want to provide an implementation of [AutoConfigurationCustomizerProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizerProvider.java), and call [addSpanExporterCustomizer](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L61-L63), something like:
```
@AutoService(AutoConfigurationCustomizerProvider.class)
public class FooCustomizer implements AutoConfigurationCustomizerProvider {
  @Override
  public void customize(AutoConfigurationCustomizer autoConfiguration) {
    autoConfiguration.addSpanExporterCustomizer(
            (spanExporter, configProperties) -> {
              if (spanExporter instanceof OtlpGrpcSpanExporter) {
                spanExporter.shutdown().join(10, TimeUnit.SECONDS);
                return ((OtlpGrpcSpanExporter) spanExporter).toBuilder().setChannel(...).build();
              }
            });
  }
}
```

Can also call do this programmatically if you're manually invoking `AutoConfiguredOpenTelemetrySdk`:
```
    AutoConfiguredOpenTelemetrySdk.builder()
            .addSpanExporterCustomizer(((spanExporter, configProperties) -> {
              ...
            }))
            .build(); 
```

**Comment by jack-berg** [comment]
> I tried to add a SpanExporterCustomizer without success.

Oh I see you tried my recommendation above. What was the problem?

**Comment by fax4ever** [comment]
> > It seems to be possible replacing the usages of opentelemetry-exporter-sender-okhttp with opentelemetry-exporter-sender-jdk and opentelemetry-exporter-sender-grpc-managed-channel.
> 
> That's correct. Based on your source code, you're using the gRPC version. The only alternative to the okhttp implementation for gRPC is `opentelemetry-exporter-sender-grpc-managed-channel`. To use, include the artifact (and exclude the `opentelemetry-exporter-sender-okhttp`, then call `setChannel`.
> 
> If you want to do this with autoconfigure, you'll want to provide an implementation of [AutoConfigurationCustomizerProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizerProvider.java), and call [addSpanExporterCustomizer](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L61-L63), something like:
> 
> ```
> @AutoService(AutoConfigurationCustomizerProvider.class)
> public class FooCustomizer implements AutoConfigurationCustomizerProvider {
>   @Override
>   public void customize(AutoConfigurationCustomizer autoConfiguration) {
>     autoConfiguration.addSpanExporterCustomizer(
>             (spanExporter, configProperties) -> {
>               if (spanExporter instanceof OtlpGrpcSpanExporter) {
>                 spanExporter.shutdown().join(10, TimeUnit.SECONDS);
>                 return ((OtlpGrpcSpanExporter) spanExporter).toBuilder().setChannel(...).build();
>               }
>             });
>   }
> }
> ```
> 
> Can also call do this programmatically if you're manually invoking `AutoConfiguredOpenTelemetrySdk`:
> 
> ```
>     AutoConfiguredOpenTelemetrySdk.builder()
>             .addSpanExporterCustomizer(((spanExporter, configProperties) -> {
>               ...
>             }))
>             .build(); 
> ```

Thanks @jack-berg I'll try to do that

**Comment by fax4ever** [comment]
> > I tried to add a SpanExporterCustomizer without success.
> 
> Oh I see you tried my recommendation above. What was the problem?

Probably I'm doing something wrong. I think I tried both the options here => https://github.com/fax4ever/telemetry-play/tree/main/sdk-autoconfigure-exporter

**Comment by jack-berg** [comment]
I finally got around to taking a closer look at this an you are totally right that something is wrong:

- If you include `opentelemetry-exporter-sender-grpc-managed-channel` and exclude `opentelemetry-exporter-sender-okhttp`, autoconfigure will fail to initialize because `opentelemetry-exporter-sender-grpc-managed-channel` expects the user to call `setChannel`, and there's no chance to do that before the span exporter customizer gets called, so we get a NPE.
- The NPE occurs because [UpstreamGrpcSenderProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/sender/grpc-managed-channel/src/main/java/io/opentelemetry/exporter/sender/grpc/managedchannel/internal/UpstreamGrpcSenderProvider.java) doesn't have any way to accommodate a null `managedChannel` argument. 
- I'm pushing a fix in #6110 which updates UpstreamGrpcSenderProvider to initialize a minimal managed channel is none is explicitly set by the user. This will allow autoconfigure to initialize without error, but still requires the user to use a span exporter customer to customize any of the options of the managed channel.
- I pushed a [commit](https://github.com/jack-berg/opentelemetry-java-docs/commit/b1b11a3ad7451d35d2cd138d9468c171cbf5b744) which demonstrates the end to end usage. Note this depends on ##6110 being merged.

**Comment by fax4ever** [comment]
> I finally got around to taking a closer look at this an you are totally right that something is wrong:
> 
> * If you include `opentelemetry-exporter-sender-grpc-managed-channel` and exclude `opentelemetry-exporter-sender-okhttp`, autoconfigure will fail to initialize because `opentelemetry-exporter-sender-grpc-managed-channel` expects the user to call `setChannel`, and there's no chance to do that before the span exporter customizer gets called, so we get a NPE.
> * The NPE occurs because [UpstreamGrpcSenderProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/sender/grpc-managed-channel/src/main/java/io/opentelemetry/exporter/sender/grpc/managedchannel/internal/UpstreamGrpcSenderProvider.java) doesn't have any way to accommodate a null `managedChannel` argument.
> * I'm pushing a fix in [Use minimal fallback managed channel when none is specified #6110](https://github.com/open-telemetry/opentelemetry-java/pull/6110) which updates UpstreamGrpcSenderProvider to initialize a minimal managed channel is none is explicitly set by the user. This will allow autoconfigure to initialize without error, but still requires the user to use a span exporter customer to customize any of the options of the managed channel.
> * I pushed a [commit](https://github.com/jack-berg/opentelemetry-java-docs/commit/b1b11a3ad7451d35d2cd138d9468c171cbf5b744) which demonstrates the end to end usage. Note this depends on #[Use minimal fallback managed channel when none is specified #6110](https://github.com/open-telemetry/opentelemetry-java/pull/6110) being merged.

Thank you @jack-berg. It does make sense to me.
